### ISSUE TITLE
Could not connect to native MeCab component

### ISSUE THREAD

**Issue by Nobiel** [siikamiika/yomichan-mecab-installer#3]
Hi,

I installed MeCab for Yomichan successfully following the [instructions](https://github.com/siikamiika/yomichan-mecab-installer/blob/master/README.md). The result after running `install_mecab_for_yomichan.py` was the same as what is shown in the screen shot. 

However, the test failed in Yomichan settings and the error message was `Could not connect to native MeCab component`. MeCab was also not enabled on the search page. I'm using Python3 and Chrome on MacOS Big Sur.

**Comment by siikamiika** [comment]
Hi, sorry for the wait. Did you manage to fix this issue? I don't have a Mac with Big Sur to test but I can maybe set up a VM at some point. Last time I tested this worked on Catalina / Chrome.

**Comment by Nobiel** [comment]
No, I didn't work out a solution. Both Firefox and Chrome throw the same error. Maybe Mecab was not installed properly but I was strictly following the instruction and Mecab is runing fine in the terminal...

**Comment by kirawi** [comment]
I also have this on Windows. I installed mecab via the installer provided on Taku's website, and extracted this script into a dev directory. I installed for Chrome and unidic, but when trying to use it with yomichan I get the above error. It's not clear what, if anything, should be in `PATH` (I added the mecab executables).

**Comment by poteau** [comment]
Did anyone manage to fix this? I'm using macOS 12.5.1 with Firefox.

**Comment by NonVoxPopuli** [comment]
I found a workaround:

It looks like line 104 of `mecab.py` isn't referencing the `mecab` in `$PATH`. 
I got around this by changing line 104 to `return './mecab'`
then symlinking `mecab` to that directory with: `ln -s $(which mecab)`

Alternatively, replacing the 'mecab' in line 104 with the output of `which mecab` also worked.

I was able to confirm the 'test' button in yomichan says "Connection was successful", and verify things parse differently, but I can't say whether things will break unexpectedly later; if someone could implement the symlink in python (and make sure it doesn't break anything) that would be a better solution.

This was on macOS 12.6.1, Firefox 106.0, and Python 3.10.8.

**Comment by poteau** [comment]
> I got around this by changing line 104 to `return './mecab'`
> then symlinking `mecab` to that directory with: `ln -s $(which mecab)`

This fixed it for me. Thanks.

**Comment by watatomo** [comment]
I'm on Windows 11 22H2 using Firefox Developer Edition. The `which mecab` method worked for me. Many thanks!

**Comment by zirong679** [comment]
> > I got around this by changing line 104 to `return './mecab'`
> > then symlinking `mecab` to that directory with: `ln -s $(which mecab)`
> 
> This fixed it for me. Thanks.

I tried to do so for my mac but it did not work for me. I understood I had to change line 104 in `mecab.py` to `return './mecab'` but I did not understand how the symlink is supposed to work. I tried pasting and running `ln -s $(which mecab)` in my terminal but it did not solve the issue when I hit test in Yomichan.

**Comment by lunacodes** [comment]
Has anyone managed to solve this on Chromebook? I tried the solutions mentioned above, in Chrome, and additionally tried installing in a shared directory, but no luck. Wondering if I'm just out of luck, because of the way ChromeOS compartmentalizes Linux.

**Comment by BernhardValenti** [comment]
having the same issue, is there any log/debug output i can see to figure out what's happening?

**Comment by Whitesttax** [comment]
Would someone mind explaining how to get this? "Alternatively, replacing the 'mecab' in line 104 with the output of which mecab also worked."
I get nothing when I run `which mecab`

**Comment by judayjude** [comment]
Had the same issue on Windows 10.  Here's what I found:

The install script does 2 main things on a Windows system:

1. downloads a dictionary to the `data` folder - creates this data folder first
2. adds a registry key - so chrome can know where to send parsing messages for yomichan

On my system, both of those tasks were silently failing.  No `data` folder created (and no dictionary download, plus no registry key created... oddly the json file that contains the details for chrome to use was created, but chrome doesn't know about it (since the registry key is missing)

Running powershell as administrator apparently fixed the first issue.  `data` directory was created and dictionary downloaded. yay!

But the registry key was still failing to be created in the windows registry...

So, I manually created the key, as per the specification in the install script

`HKEY_CURRENT_USER\SOFTWARE\Google\Chrome\NativeMessagingHosts\yomichan_mecab`

...and set the key's value to the path for where I had my `mecab_yomichan.json` file.

Everything works now, as verified by the Test button in the yomichan config page!

Hope that helps someone.

**Comment by sonbui00** [comment]
You can see my fix for MacOS - Chrome
https://github.com/themoeway/yomitan-mecab-installer/pull/4
